### PR TITLE
filebeat/module/{iis,o365}: Fix uri_parts ingest pipeline processor output wrong extension

### PR DESCRIPTION
**Pull request by muthu-mps**
## Proposed commit message

This PR fixes the errors in the pipeline tests which generates incorrect values for the `extension` field of `uri_parts` processor. [Bug fix](https://github.com/elastic/elasticsearch/issues/105612) done in Elasticsearch to remove the incorrect url.extension values from the uri_parts ingest pipeline processor. The affected beats module should remove this field from the generated sample documents. IIS error module and the o365 audit log modules are effected by this change.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally
cd beats/filebeat
GENERATE=true TESTING_FILEBEAT_MODULES=iis,o365 mage -v pythonIntegTest

## Related issues
This fixes the python integration tests failure reported in the below [PR](https://github.com/elastic/beats/pull/38122)

### CODE REVIEW

**Review by leehinman — APPROVED** [review]
LGTM

**Review by ShourieG — APPROVED** [review]
Lgtm

**Review by shmsr — APPROVED** [review]

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -100,6 +100,7 @@ fields added to events containing the Beats version. {pull}37553[37553]
 - Prevent HTTPJSON holding response bodies between executions. {issue}35219[35219] {pull}38116[38116]
 - Fix "failed processing S3 event for object key" error on aws-s3 input when key contains the "+" character {issue}38012[38012] {pull}38125[38125]
 - Fix duplicated addition of regexp extension in CEL input. {pull}38181[38181]
+- Fix the incorrect values generated by url_parts processor. {pull}38216[38216]
```
Comment (shmsr) [inline CHANGELOG.next.asciidoc:103]:
```suggestion
- Fix the incorrect values generated by the uri_parts processor. {pull}38216[38216]
```

**Review by shmsr — CHANGES_REQUESTED** [review]
Left one comment.

**Review by shmsr — APPROVED** [review]

**Pull request merged**